token/js/test/unit/transferHook.test.ts

```diff
@@ -0,0 +1,139 @@
+import { getExtraAccountMetas, resolveExtraAccountMeta } from '../../src';
+import { expect } from 'chai';
+import { AccountMeta, Keypair, PublicKey } from '@solana/web3.js';
+
+describe('transferHookExtraAccounts', () => {
+    const testProgramId = new PublicKey('7N4HggYEJAtCLJdnHGCtFqfxcB5rhQCsQTze3ftYstVj');
+    const instructionData = Buffer.from(Array.from(Array(32).keys()));
+    const plainAccount = new PublicKey('6c5q79ccBTWvZTEx3JkdHThtMa2eALba5bfvHGf8kA2c');
+    const pdaPublicKey = new PublicKey('8Zibh9ywkKCZc3h23vwxvdMPG9yhg4LYYzky2yRkLHBJ');
+    const pdaPublicKeyWithProgramId = new PublicKey('9Apejge9XvJ3toZ8PX2sKKgMoyN7sSegzdq8n1kMzRae');
```

Comment:
To give some more confidence in the test, can you derive these PDAs dynamically from the inputs rather than putting them in statically?